src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -39,417 +28,560 @@
-				//and did not recognise the URL, so redirected to a login page, for instance
-				//Remediation: Check the contents match the expected third party contents.
-				if (log.isDebugEnabled()) log.debug("Raw Socket established, in theory to "+thirdpartyHost);
-				
-				OutputStream os = socket.getOutputStream();
-				InputStream is = socket.getInputStream();
-				
-				PrintWriter pw = new PrintWriter(os, false);
-				pw.write("GET http://"+thirdpartyHost + ":"+thirdpartyPort + "/ HTTP/1.1\n");
-				pw.write("Host: "+thirdpartyHost+"\n\n");
-				pw.flush();
-				
-				//read the response via a 4k buffer
-				ByteArrayOutputStream bos = new ByteArrayOutputStream ();
-				byte [] buffer = new byte [1024*4];
-				int bytesRead = is.read(buffer);
-				int totalBytesRead = 0;
-				while (bytesRead > -1) {
-					totalBytesRead+= bytesRead;					
-					bos.write(buffer, 0, bytesRead);					
-					bytesRead = is.read(buffer);
-				}
-				String response = new String (bos.toByteArray());				
-				if (log.isDebugEnabled()) log.debug("Response is "+totalBytesRead+" bytes: \n"+response);
-				
-				Matcher m = thirdPartyContentPattern.matcher(response);
-				if (m.matches()) {
-					if (log.isDebugEnabled()) log.debug("Response matches expected third party pattern!");
-					is.close();
-					os.close();
-					bos.close();
-					socket.close();
-					return true;
-				} else { 
-					if (log.isDebugEnabled()) log.debug("Response does *not* match expected third party pattern");
-				}
-				is.close();
-				os.close();
-				bos.close();
-				socket.close();
-				return false;				
-			}
-			else  {
-				if (log.isDebugEnabled()) { 
-					log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
-					log.debug ("The status line returned: "+statusLine);
-				}
-				return false;
-			} 
-		}
-		catch (Exception e) {
-			if (log.isDebugEnabled()) log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
-		}
-		return false;
-	}
+    /* These are the 'default' HTTP methods which are considered as insecure */
+    private static final List<String> INSECURE_DEFAULT_METHODS = Arrays.asList(
+//            /* Used by proxy server */
+//            HttpRequestHeader.CONNECT,
+            /* This is almost never implemented by the server */
+            HttpRequestHeader.DELETE,
+            /* Request for information about the communication options available */
+            HttpRequestHeader.OPTIONS,
+            /* Request for a change in a resource identified by the URI */
+            HttpRequestHeader.PATCH,
+            /* For putting or updating a resource on the server */
+            HttpRequestHeader.PUT
+            /* Mostly used for debugging purposes */
+//            HttpRequestHeader.TRACE,
+//            /* M$ version of TRACE */
+//            HttpRequestHeader.TRACK
```

Comment:
Why have CONNECT, TRACE, and TRACK been commented out?

Reply:
Because they're tested separately by separate methods I have no idea why the previous contributer added them in the same list if they're being tested individually.